### DISCUSSION TITLE
Specifying rendering intent when converting via -profile command

### DISCUSSION THREAD

**Discussion by BrooksWait** [ImageMagick/ImageMagick#2858] [Help] [answered]
Hi,
I am using the ImageMagick Convert utility to convert image pixel values via an embedded color profile.  I am using the -profile  command as detailed below.  This command is working properly and produces the desired converted image.

"[absolute path]\convert.exe" "0534-0_assigned.jpg" -profile "[absolute path]\Adobe RGB (1998).icc" +profile * "[absolute path]\0534-0_converted_absolute.jpg"

Now, I would like to specify a rendering intent also.  Based on the documentation I have found, I would assume that this would be as follows:

"[absolute path]\convert.exe" "0534-0_assigned.jpg" -profile "[absolute path]\Adobe RGB (1998).icc" **-intent Absolute** +profile * "[absolute path]\0534-0_converted_absolute.jpg"

This command does not error and the desired image file is produced.  However, I have created images for all rendering intents (Relative, Perceptual, etc.) and then compared the pixel values of these images and there is no difference.  This makes me think that the rendering intent may not be working properly?

Does anyone know if I am specifying the "-intent Absolute" corrrectly?  Any help would be appreciated.

Thanks in advance!
BWait

**Comment by snibgo** [comment, marked as answer]
I assume your input image 0534-0_assigned.jpg already has an embedded profile, so you are converting to a new profile.

"-intent" works fine, as far as I know. However, not all ICC profiles support all intents. For example, a profile that works by look-up tables (rather than formulae) can't use different methods for out-of-gamut colours. The absolute and relative intents may be equal, and the perceptual and saturation intents may be equal.

**Reply by BrooksWait** [reply]
Thanks for your response snibgo!

BWait
